src/needs-parens.js

```diff
@@ -92,8 +91,6 @@ function needsParens(path) {
           }
 
           if (pp === np && key === "right") {
-            assert.strictEqual(parent.right, node);
```

Comment:
This actually meanless, since `key === "right"` already ensure it's accessing `parent.right`.

Reply:
I mean, I kind of fear that we could be running into a Chesterton's Fence problem here. But I don't see any case where that line would be needed either. 🤷‍♂️